micro_sam/training/util.py

```diff
@@ -210,6 +224,29 @@ def __call__(self, x, y, n_pos, n_neg, get_boxes=False, n_samples=None):
         return batched_inputs, batched_sampled_cell_ids_list
 
 
+class ConvertToSemanticSamInputs:
+    """
+    """
+    def __call__(self, x, y):
+        """Convert the outputs of dataloader to the batched format of inputs expected by SAM.
+        """
+        batched_inputs, gt_logits = [], []
+        for image, gt in zip(x, y):
+            batched_input = {"image": image, "original_size": image.shape[1:]}
+            batched_inputs.append(batched_input)
+
+            # downsize the labels
+            gt_shape = (gt.shape[0], 256, 256)
+            gt_logits.append(
+                resize(image=gt, output_shape=gt_shape, preserve_range=True, order=0, anti_aliasing=False)
```

Comment:
This will change the dtype. I am not sure if this is a problem though. It can be prevented by adding `.asyupe(gt.dtype)` at the end.